kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/LocallyRunningOperatorRbacHandler.java

```diff
@@ -0,0 +1,242 @@
+            .withNewMetadata()
+                .withName("frameowork-cluster-role-binding")
+            .endMetadata()
+            .withNewRoleRef()
+                .withName(frameworkClusterRole.getMetadata().getName())
+                .withKind(frameworkClusterRole.getKind())
+                .withApiGroup("rbac.authorization.k8s.io")
+            .endRoleRef()
+            .addNewSubject()
+                .withName(impersonatedUser)
+                .withKind("User")
+                .withApiGroup("rbac.authorization.k8s.io")
+            .endSubject()
+            .build();
+    // @formatter:on
+
+    @Override
+    public void beforeEach(ExtensionContext context) {
+        try (var adminClient = OperatorTestUtils.kubeClient();
+                var files = Files.list(Path.of("install"))) {
+            files.sorted().forEach(resourceFile -> {
+                try {
+                    var list = adminClient.load(new FileInputStream(resourceFile.toString())).items();
+                    list.stream()
+                            .filter(this::isClusterRoleOrRoleBindingResource)
+                            .map(r -> {
+                                // We need to rewrite the RBAC so that rather than being in terms
+                                // of a ServiceAccount, they are in terms of our impersonating user.
+                                if (r instanceof ClusterRoleBinding crb) {
+                                    crb.getSubjects().stream()
+                                            .filter(s -> s.getKind().equals("ServiceAccount"))
+                                            .forEach(s -> {
+                                                var originalUser = s.getName();
+                                                s.setKind("User");
+                                                s.setNamespace(null);
+                                                s.setName(impersonatedUser);
+                                                s.setApiGroup("rbac.authorization.k8s.io");
+                                                LOGGER.trace("Updated cluster role binding for {} to match impersonated user {}", originalUser,
+                                                        impersonatedUser);
+                                            });
+                                }
+                                return r;
+                            })
+                            .forEach(r -> {
+                                LOGGER.trace("Creating/patching: {} from {}", r.getMetadata().getName(), resourceFile);
+                                adminClient.resource(r).createOr(EditReplacePatchable::patch);
+                            });
+                }
+                catch (FileNotFoundException e) {
+                    throw new UncheckedIOException("failed to install cluster resources " + resourceFile, e);
+                }
+            });
+
+            // The test framework itself needs these roles.
```

Comment:
Gah - this is nasty.   Ideally we'd want a way to give the LocallyRunOperatorExtension two clients - one to use for its own setup purposes and one to actually give to the operator under test.  I'll raise an issue against JOSDK (assuming you think I'm on a reasonable path).

Reply:
I think it'd be useful if upstream supported testing RBAC like this, it's going to be so much quicker than building up images and finding out the RBAC is mismatched.

Introducing the split clients in the extension sounds like a good first step.

Maybe the local extension could later be responsible for generating the operator subject name and the user could supply some function that maps `subjectName -> operatorResources: Set<CustomResource>`. It would be nice if the extension was handling this apply/delete logic of these additional resources.